sdk/inc/azure/core/az_json.h

```diff
@@ -792,6 +792,27 @@ AZ_NODISCARD az_result az_json_reader_next_token(az_json_reader* ref_json_reader
  */
 AZ_NODISCARD az_result az_json_reader_skip_children(az_json_reader* ref_json_reader);
 
+/**
+ * @brief Gets the JSON token's string after unescaping it, if required.
+ *
+ * @param[in] json_string The #az_span that contains the string to be unnescaped.
+ * @param destination Pointer to the buffer that will contain the oputput.
```

Comment:
It's not clear here that the function allows _in-place_ unescaping.
I liked the idea of having 2 APIs (one in-place and one generic) to indicate setting the `json_string` and `destination` to point at the same memory location.

Reply:
@ahsonkhan didn't want that in order to reduce the API surface. Apparently that API does not meet the requirements. And since it was a wrapper, and the test show how to do inplace i reluctantly agreed.